plugin.py

```diff
@@ -132,6 +148,116 @@ def run(self, edit: sublime.Edit) -> None:
         session.send_request(Request("rust-analyzer/reloadWorkspace"))
 
 
+class RustAnalyzerMemoryUsage(LspTextCommand):
+    session_name = "rust-analyzer"
+
+    def run(self, edit: sublime.Edit) -> None:
+        session = self.session_by_name(self.session_name)
+        if session is None:
+            return
+
+        session.send_request(Request("rust-analyzer/memoryUsage"), self.on_result)
+
+    def on_result(self, payload: Optional[str]) -> None:
+        view = sublime.active_window().new_file()
+        view.set_scratch(True)
+        view.set_name("--- RustAnalyzer Memory Usage ---")
+        view.run_command("append", {"characters": "Per-query memory usage:"})
+        for line in payload.splitlines():
+            view.run_command("append", {"characters": '\n{}'.format(line)})
+        view.run_command("append", {"characters": "\n(note: database has been cleared)"})
+        view.set_read_only(True)
+
+
+
+
+class RustAnalyzerExec(LspTextCommand):
+    session_name = "rust-analyzer"
+
+    def run(self, edit: sublime.Edit) -> None:
+        params = text_document_position_params(self.view, self.view.sel()[0].b)
+        session = self.session_by_name(self.session_name)
+        if session is None:
+            return
+        session.send_request(Request("experimental/runnables", params), self.on_result)
+
+    def run_termius(self, check_phrase: str , payload) -> None:
+        output = None
+        for i in range(len(payload)):
+            if payload[i]["label"].startswith(check_phrase):
+                output = payload[i]
+
+        if not output:
+            return
+
+        if not Terminus:
+            sublime.error_message(
+                'Cannot run executable "{}": You need to install the "Terminus" package and then restart Sublime Text'.format(output["kind"]))
+            return
+
+        cargo_path = '"{}"'.format(shutil.which("cargo"))
+        run_command = [cargo_path] + output["args"]["cargoArgs"] + output["args"]["cargoExtraArgs"] + output["args"]["executableArgs"]
+        cmd = " ".join(run_command)
+        args = {
+            "title": output["label"],
+            "shell_cmd": cmd,
+            "cwd": output["args"]["workspaceRoot"],
+            "auto_close": get_setting(self.view, "terminus_auto_close", False)
+        }
+        if get_setting(self.view, "terminus_use_panel", False):
+            args["panel_name"] = output["label"]
+        window = self.view.window()
+        window.run_command("terminus_open", args)
```

Comment:
```suggestion
        window = self.view.window()
        if window is None:
            return

        window.run_command("terminus_open", args)
```

Probably might be worth moving this `None` check to the top.